🔧 Fixed missing dependencies for additional file parsers in Dockerfile and some refactoring

SpaceShaman opened the pull request:
This update introduces docker-compose support for easier project setup and deployment.

## Added

### Docker Support
  - Added a `docker-compose.yml` file to simplify running the project in a Docker environment.

## Refactor
  - Renamed .env to .env.default for improved security and to prevent accidental exposure of sensitive information. Updated documentation to reflect the new environment configuration process.
  - Included Gradio server settings in the `.env.default` file for easier configuration.

## Documentation
  - Updated the README with instructions for using `docker-compose` and improved command syntax highlighting.

## Fixed
  - Added missing dependencies for additional file parsers in Dockerfile
  - Added unstructured[all-docs] to pip install for extended document support

Without these dependencies, various types of errors appeared when trying to parse documents in a format other than PDF

SpaceShaman committed:
🔧 (env): add Gradio server settings to .env file
🔧 (docker-compose.yml): add docker-compose configuration for the project

SpaceShaman committed:
📝 (README.md): update Docker command syntax highlighting
📝 (README.md): add instructions for running with docker-compose

SpaceShaman committed:
📝 (README.md): specify shell syntax highlighting for code block

SpaceShaman committed:
🔧 (config): add .env.default file with default environment variables
📝 (README): update instructions to use .env.default for environment variables
📝 (README): correct docker-compose to docker compose for consistency
🔧 (gitignore): remove .env from .gitignore to allow environment configuration

SpaceShaman committed:
🔥 (config): remove .env file to avoid exposing sensitive information

SpaceShaman committed:
🔧 (gitignore): add .env file to .gitignore to prevent sensitive data from being committed

SpaceShaman committed:
🔧 (Dockerfile): remove experimental syntax directive to simplify Dockerfile
🔧 (Dockerfile): add missing dependencies for additional file parsers
⬆️ (Dockerfile): add unstructured[all-docs] to pip install for extended document support

SpaceShaman committed:
🔧 (gitignore): add .ktem_app_data/ to ignore application data directory

phv2312 commented:
@SpaceShaman thanks for your contribution. From my side I think your PR is quite good. 
I just have two concerns, related to docker and may need @taprosoft to consult us as well:

- The entrypoint in Dockerfile, should we change from `gradio launch.py` to `python launch.py`. The 1st command from my point of view should be used while under developing only.
- I see you uncomment some apts such as ffmpeg, tesseract, ... to install unstructure. This can handle more file types but the trade off is increasing the docker image size (IF I remember correctly from my last experiment in will add around 2Gb). @taprosoft Can you help to make decision for this point ?

SpaceShaman commented:
@phv2312
> I see you uncomment some apts such as ffmpeg, tesseract, ... to install unstructure. This can handle more file types but the trade off is increasing the docker image size (IF I remember correctly from my last experiment in will add around 2Gb). @taprosoft Can you help to make decision for this point ?

In addition to uncommenting the apts dependency, I also added `libmagic-dev` there and added `unstructured[all-docs]` to the pip installation, because without it I couldn't add csv and txt files.

We could use multi-stage building to define a basic version and an extended version with additional dependencies for other file types, like below:

```dockerfile
FROM python:3.10-slim as basic
ENV PYTHONDONTWRITEBYTECODE=1
ENV PYTHONUNBUFFERED=1
ENV PYTHONIOENCODING=UTF-8
WORKDIR /app

RUN apt update -qqy \
  && apt install -y \
  ssh git \
  gcc g++ \
  poppler-utils \
  libpoppler-dev \
  && \
  apt-get clean && \
  apt-get autoremove

COPY . /app

RUN --mount=type=ssh pip install -e "libs/kotaemon[all]"
RUN --mount=type=ssh pip install -e "libs/ktem"
RUN pip install graphrag future
RUN pip install "pdfservices-sdk@git+https://github.com/niallcm/pdfservices-python-sdk.git@bump-and-unfreeze-requirements"

ENTRYPOINT ["python", "app.py"]

FROM basic as full

RUN apt install -y \
  tesseract-ocr \
  tesseract-ocr-jpn \
  libsm6 \
  libxext6 \
  ffmpeg \
  libmagic-dev

RUN pip install unstructured[all-docs]
```
